lib/src/_network_image_io.dart

```diff
@@ -339,19 +182,195 @@ class ExtendedNetworkImageProvider
   Future<Uint8List?> getNetworkImageData({
     StreamController<ImageChunkEvent>? chunkEvents,
   }) async {
-    final String uId = cacheKey ?? keyToMd5(url);
+    return await _(url, chunkEvents);
+  }
 
-    if (cache) {
-      return await _loadCache(
-        this,
-        chunkEvents,
-        uId,
+  Future<Directory> _getCacheDir() async {
+    final Directory dir = Directory(
+        join((await getTemporaryDirectory()).path, cacheImageFolderName));
+    // ignore: avoid_slow_async_io
+    if (!await dir.exists()) {
```

Comment:
Maybe the synchronized method?

```suggestion
    if (!dir.existsSync()) {
```